### DISCUSSION TITLE
Random disconnect problem solution

### DISCUSSION THREAD

**Discussion by drcyberg** [strongswan/strongswan#1859] [Ideas]
Hi All and @tobiasbrunner !

It looks like there is a solution to the random VPN disconnect.

**[ANTECEDENT]**
No matter what I set within the program, it always disconnected randomly. After a long search, I came across a description that theoretically solves this problem.

**[SOLUTION]**
Each router has an Offloading solution (SW/HW) and IPsec Offloading Support: [LINK](https://help.ui.com/hc/en-us/articles/115006567467-EdgeRouter-Hardware-Offloading)
And this part gets stuck and a problem occurs, especially if it is under load (many clients are connected). IKE_AUTH response packets don't get through.
And the correct setting of the ESP is also important from the router's point of view (CPU/MCU Support).

I hope this solves the disconnection problem. :-)

Regards: DrCyberg

**Comment by tobiasbrunner** [comment]
> IKE_AUTH response packets don't get through.

Why would that be related to IPsec offloading? IKE != IPsec/ESP.

**Reply by drcyberg** [reply]
Yes, this is also necessary the IKE/ESP selector in strongswan app between client (linux kernel), oflload must be enabled on device (MCU), and check IPsec Offloading support on router.
If everyone already knew this requirement, I'm sorry, but I understood it now. :-)

**Reply by tobiasbrunner** [reply]
I didn't and I don't understand.

**Reply by drcyberg** [reply]
EdgeRouter - Hardware Offloading: `"...Not all available ESP hashing/encryption algorithms are compatible with offloading. IKE traffic is not offloaded but this is only used to establish the tunnel and will not affect the performance..."`
Here there was a problem with effective ESP traffic, and this was exacerbated by the fact that offloading was not used/disabled.
I think this should also be taken into account before setting the encryption level.
Since we have been using it this way, the VPN connection has been stable and fast.

**Reply by tobiasbrunner** [reply]
OK, having to configure a specific ESP proposal makes sense if only some algorithms are supported if offloading is enabled. But I still don't get the connection to IKE and your claim of problems with IKE_AUTH getting stuck being related to this. How can this affect the IKE traffic? 

Or did you rather mean that the IKE_AUTH exchange was failing (e.g. with a NO_PROPOSAL_CHOSEN error) because the CHILD_SA couldn't get installed because of the incompatible algorithms?

**Reply by drcyberg** [reply]
Bingo, the conversion has already taken place, but it actually froze many times due to incompatibility, i.e. IKE_AUTH exchange failed. I tried to play tricks by changing various parameters with the `ipsec.conf` file, but it was not satisfactory in the long run.
It's about this _"wonderful"_ (old)router: [Ubiquiti USG](https://store.ui.com/us/en/products/usg)

Even with older routers, it will work stably with the most basic ESP settings (maybe): `3DES`
Above that, the connection is already unstable.

**Comment by drcyberg** [comment]
@tobiasbrunner I didn't want to create separate posts because of this, so I thought I'd continue here.

**UPDATE - 1**
An error occurred that when accessing several different VPN networks with 1 internet connection, the communication in the channel is randomly interrupted (the connection is still active).
In the case of a company, it is inevitable that several people use VPN connections, of course if it is permitted by law.

**SOLUTION**
When setting the firewall rule, the IP address of the remote PC must be entered, so that 500,4500 UDP packets flow in a regulated manner. if not specified, interference can easily occur in the channel.

**Comment by drcyberg** [comment]
UPDATE - 2
`Rekeying CHILD_SAs is also supported by the Windows client. For some reason, a client behind NAT does not accept a rekeying attempt and rejects it.` [LINK](https://docs.strongswan.org/docs/5.9/interop/windowsClients.html#_child_sa_rekeying)

**SOLUTION**
When setting up a Windows Firewall rule, there is an option to allow computers to dynamically negotiate encryption.
`This setting enables the sending of authenticated but unencrypted network packets during encryption negotiation.` [LINK](https://learn.microsoft.com/en-us/powershell/module/netsecurity/new-netfirewallrule?view=windowsserver2022-ps#-encryption)
